plugins/modules/custom_image.py

```diff
@@ -0,0 +1,314 @@
+    state: absent
+'''
+
+RETURN = r'''
+href:
+  description: The API URL to get details about this resource.
+  returned: success when state == present
+  type: str
+  sample: https://api.cloudscale.ch/v1/custom-imges/11111111-1864-4608-853a-0771b6885a3a
+uuid:
+  description: The unique identifier of the custom image.
+  returned: success
+  type: str
+  sample: 11111111-1864-4608-853a-0771b6885a3a
+name:
+  description: The human readable name of the custom image.
+  returned: success
+  type: str
+  sample: alan
+created_at:
+  description: The creation date and time of the resource.
+  returned: success
+  type: str
+  sample: "2020-05-29T13:18:42.511407Z"
+slug:
+  description: A string identifying the custom image for use within the API.
+  returned: success
+  type: str
+  sample: foo
+checksums:
+  description: The checksums of the custom image as key and value pairs. The
+    algorithm (e.g. sha256) name is in the key and the checksum in the value.
+    The set of algorithms used might change in the future.
+  returned: success
+  type: dict
+  sample: {
+    "md5": "5b3a1f21cde154cfb522b582f44f1a87",
+    "sha256": "5b03bcbd00b687e08791694e47d235a487c294e58ca3b1af704120123aa3f4e6"
+  }
+user_data_handling:
+  description: How user_data will be handled when creating a server. There are
+    currently two options, "pass-through" and "extend-cloud-config".
+  returned: success
+  type: str
+  sample: "pass-through"
+tags:
+  description: Tags assosiated with the custom image.
+  returned: success
+  type: dict
+  sample: { 'project': 'my project' }
+status:
+  description: Shows the progress of an import. Values are one of
+    "in_progress", "success" or "error".
+  returned: success
+  type: str
+  sample: "in_progress"
+state:
+  description: The current status of the custom image.
+  returned: success
+  type: str
+  sample: present
+'''
+
+from ansible.module_utils.basic import (
+    AnsibleModule,
+)
+from ..module_utils.api import (
+    AnsibleCloudscaleBase,
+    cloudscale_argument_spec,
+)
+
+
+class AnsibleCloudscaleCustomImages(AnsibleCloudscaleBase):
+
+    def query(self):
+
+        # Initialize
+        self._resource_data = self.init_resource()
+
+        # Set import status
+        import_status = self._module.params.pop('import_status', False)
+
+        # Query by UUID
+        uuid = self._module.params[self.resource_key_uuid]
+
+        if uuid is not None:
+
+            # Get image by uuid
+            resource = self._get('%s/%s' % (self.resource_name, uuid))
+
+            updated = False
+            if resource:
+                # Check if slug, tags or user_data_handling needs update
+                for key in ['slug', 'tags', 'user_data_handling']:
+                    if self._module.params[key] is not None:
+                        if self.find_difference(key,
+                                                resource,
+                                                self._module.params[key]):
```

Comment:
btw ansible's max line limit is 160, just saying

Reply:
I generally nag people to use 79, and @ctx has had to pass a few Python reviews of mine already 😉